### PR TITLE
[vernac] Split `command.ml` into separate files.

### PR DESCRIPTION
**Pull request by ejgallego**
Over the time, `Command` grew organically and it has become now one of
the most complex files in the codebase; however, its functionality is
well separated into 4 key components that have little to do with each
other.

We thus split the file, and also document the interfaces. Some parts
of `Command` export tricky internals to use by other plugins, and it
is common that plugin writers tend to get confused, so we are more
explicit about these parts now.

We also remove the dependency on the `program_mode` global flag, now the mode is an attribute of the commands that support it.

This patch depends on #6413.

### CODE REVIEW

File: dev/base_include [hunk 1]
```diff
@@ -170,7 +170,7 @@ open Eqschemes
 
 open ExplainErr
 open Class
-open Command
+open ComDefinition
```
Comment (SkySkimmer) [inline dev/base_include]: This should be in preceding commit.

File: vernac/vernacentries.ml [hunk 2]
```diff
@@ -592,18 +591,29 @@ let vernac_inductive ~atts cum lo finite indl =
...
     in
     let indl = List.map unpack indl in
-    do_mutual_inductive indl is_cumulative atts.polymorphic lo finite
+    ComInductive.do_mutual_inductive indl is_cumulative atts.polymorphic lo finite
 
 let vernac_fixpoint ~atts discharge l =
   let local = enforce_locality_exp atts.locality discharge in
   if Dumpglob.dump () then
     List.iter (fun (((lid,_), _, _, _, _), _) -> Dumpglob.dump_definition lid false "def") l;
+  (* XXX: Match the flag *)
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml]: ???

**Review by SkySkimmer — APPROVED** [review]
I'm assuming this just moves code around.

File: vernac/vernacentries.ml (hunk 2 above)
Comment (ejgallego) [reply vernac/vernacentries.ml]: This is just a reminder to switch this check to the attributes system once it lands.

File: dev/base_include (hunk 1 above)
Comment (ejgallego) [reply dev/base_include]: Good catch.

**Comment by ejgallego** [comment]
> I'm assuming this just moves code around.

For command yes, but there are a couple of tweaks in the handling of flags, semantics should be preserved thou.

**Comment by mattam82** [comment]
I fixed equations accordingly

**Pull request merged**